### DISCUSSION TITLE
How to loop into all documents from all libraries in a sitecollection?

### DISCUSSION THREAD

**Discussion by khha** [pnp/powershell#4588] [Q&A] [answered]
I would like to loop into all documents from all libraries in a sitecollection. I try below, but is not working. How to do this? I would like to avoid folders and would like to use recursive and itemtype file.

```
$libraries = Get-PnPList | Where-Object {$_.BaseType -eq "DocumentLibrary" } 

                foreach($library in $libraries) {
                    $documents = Get-PnPFolderItem -Identity $library -Recursive -ItemType File
                    foreach($document in $documents) {
                           // do something
                     }
                }
```

**Comment by Barbarur** [comment]
What if instead of using Get-PnPFolderItem, you use Get-PnPListItem?

**Reply by khha** [reply]
> What if instead of using Get-PnPFolderItem, you use Get-PnPListItem?

Hi @Barbarur , with Get-PnPListItem I cannot get only files and recursive into folders and subfolders.

**Reply by Barbarur** [reply]
You mentioned it's not working. What error are you getting?

Get-PnPListItem gets all files and folders, then you can filter out folders by:
$_.FileSystemObjectType -eq "Folder"

**Reply by khha** [reply]
@Barbarur  Some error about object type mismatch of variable at this part -Identity $library. If I give the library name it works. For example "Shared Documents". But if I use $library.Title it gives me the library title "Documents". I cannot find some property where it gives me the library name "Shared Documents"

**Reply by Barbarur** [reply]
Are you use you use $library.title and not simple $library? On the sample code you shared you use the second option.

Add some write-host in between to check the values you are using, so you can understand better from where the error is coming from.

**Reply by khha** [reply]
> $library.title and not simple $library

I try both and is not working. If I use $library I got an error that it is expecting another object type. If I try $library.Title it use the value "Document" which gives the error that he cannot find library with that name. He expect the name "Shared Documents".

But is it working from your side?

**Reply by Barbarur** [reply, marked as answer]
I've reviewed the docs. -Identity is asking for the instance of the folder, but you have been using the Title or the list itself.

You can try using $library.rootfolder

Otherwise you can trim the library url, trim it and use the parameter -FoldersiterelativeUrl
